app/src/main/java/org/mozilla/focus/settings/permissions/permissionoptions/SitePermissionOptionsMiddleware.kt

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.focus.settings.permissions.permissionoptions
+
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.MiddlewareContext
+
+class SitePermissionOptionsMiddleware(val sitePermission: SitePermission, val storage: SitePermissionOptionsStorage) :
+    Middleware<SitePermissionOptionsScreenState, SitePermissionOptionsScreenAction> {
+
+    override fun invoke(
+        context: MiddlewareContext<SitePermissionOptionsScreenState, SitePermissionOptionsScreenAction>,
+        next: (SitePermissionOptionsScreenAction) -> Unit,
+        action: SitePermissionOptionsScreenAction
+    ) {
+        when (action) {
+            is SitePermissionOptionsScreenAction.Select -> {
+                storage.saveCurrentSitePermissionOptionInSharePref(
+                    action.selectedSitePermissionOption,
+                    sitePermission = sitePermission
+                )
+                next(action)
+            }
+            is SitePermissionOptionsScreenAction.InitSitePermissionOptions -> {
```

Comment:
Handling this action here seems again to mean that the Middleware is directly asked to actually set the initial state in store which is a departure from the current way of using these on Fenix and of the Redux definition of middlewares being extensions to the app functionality while in this case it would become the core functionality - https://redux.js.org/understanding/history-and-design/middleware.

Could this initial state be set in place, when the store is created?
Otherwise what's being used on Fenix is to create the store with an empty state and then update it later from the same components where this store was created - in this case by just dispatching `UpdateSitePermissionOptions`.